docs/Changelog.md

```diff
@@ -4,6 +4,7 @@
 -------------------------
 
 * [#6008](https://github.com/TouK/nussknacker/pull/6008) Add embedded QuestDB as database for FE statistics.
+* [#6053](https://github.com/TouK/nussknacker/pull/6053) Add impersonate mechanism for technical users
```

Comment:
@Elmacioro we have to update this https://nussknacker.io/documentation/docs/installation_configuration_guide/DesignerConfiguration/#users-roles-and-permissions documentation. 
BTW please remove mention about `AdminTab` - it is some obsolete garbage.
After updating the docs, I think that this will be exposed to user. Not the configurator, but for the admin who can integrate another application with designer. WDYT @mk-software-pl ?

Reply:
I updated the doc and removed the `AdminTab` mention

Reply:
But currently, from the users' perspective, there is no impersonation support yet, right? I've checked where `AuthenticatedUser` is instantiated and I didn't find the place where we create an instance with the `impersonatedAuthenticationUser` field != None. You've mentioned the next PR. Maybe we should add this changelog entry in the next PR instead?

Reply:
Currently none of the implementations of `AuthenticationResources` (which provides `AuthenticatedUser`) use this mechanism in Nu (it will change with next PR with migrations).
However, somebody integrating another application with the designer can have their own `AuthenticationResources` implementation (for example for SSO) with impersonated user.

Reply:
Uhh, so we don't have any end to end test as well? It is not acceptable! :) With regards to our DoD, this should be a part of this change.

Reply:
Also, as we discusses, with a little bit effort we could make this impersonation logic not plugin-specific. Why not to use the same header, no matter which authentication mechanism we pick?

Reply:
Ok, I'll try to modify it so we could use the header independently of AuthenticationResources implementation. It will also make it easier to add an end to end test for this PR

Reply:
I introduced `AuthenticationManager` that coordinates anonymous access, impersonated access and plugin specific authentication with `AuthenticationResources`. 
The impersonated access should be possible by attaching `Impersonate-User-Data` header by a user with `Impersonate` permission. I added some tapir and akka based tests to check whether user gets impersonated with correct set of permissions.

Reply:
I modified the changelog entry to mention basic auth and the API